terraform/main.tf

```diff
@@ -5,155 +5,227 @@ terraform {
   }
 }
 
-module "service_accounts" {
-  source        = "terraform-google-modules/service-accounts/google"
-  version       = "~> 2.0"
-  project_id    = var.project_id
-  prefix        = var.prefix
-  generate_keys = true
-  names         = ["cd-sa"]
-  project_roles = [
-    "${var.project_id}=>roles/container.admin",
-    "${var.project_id}=>roles/artifactregistry.writer",
-    # FIXME: This is way too much perms just to ssh into a node
-    "${var.project_id}=>roles/compute.instanceAdmin.v1"
-  ]
+// Service account used by all the nodes and pods in our cluster
+resource "google_service_account" "cluster_sa" {
+  account_id   = "${var.prefix}-cluster-sa"
+  display_name = "Cluster SA for ${var.prefix}"
+  project      = var.project_id
 }
 
-output "ci_deployer_key" {
-  value     = module.service_accounts.keys["cd-sa"]
-  sensitive = true
+// To access GCS buckets with requestor pays, the calling code needs
+// to have serviceusage.services.use permission. We create a role
+// granting just this to provide the cluster SA, so user pods can
+// use it. See https://cloud.google.com/storage/docs/requester-pays
+// for more info
+resource "google_project_iam_custom_role" "identify_project_role" {
+  // Role names can't contain -, so we swap them out. BOO
+  role_id     = replace("${var.prefix}_user_sa_role", "-", "_")
+  project     = var.project_id
+  title       = "Identify as project role for users in ${var.prefix}"
+  description = "A description"
+  permissions = ["serviceusage.services.use"]
 }
 
-resource "google_artifact_registry_repository" "container_repository" {
-  provider = google-beta
-
-  location      = var.region
-  repository_id = "low-touch-hubs"
-  format        = "DOCKER"
-  project       = var.project_id
+resource "google_project_iam_member" "identify_project_binding" {
+  project = var.project_id
+  role    = google_project_iam_custom_role.identify_project_role.name
+  member  = "serviceAccount:${google_service_account.cluster_sa.email}"
 }
 
-// Give the GKE service account access to our artifact registry docker repo
-resource "google_project_iam_member" "project" {
+resource "google_project_iam_member" "cluster_sa_roles" {
+  for_each = var.cluster_sa_roles
+
   project = var.project_id
-  role    = "roles/artifactregistry.reader"
-  member  = "serviceAccount:${module.gke.service_account}"
+  role    = each.value
+  member  = "serviceAccount:${google_service_account.cluster_sa.email}"
 }
 
+resource "google_container_cluster" "cluster" {
+  # config_connector_config is in beta
```

Comment:
How beta is this beta? I know you like to live on the edge :wink:
Just trying to understand how extensively used the beta is by the community.
I see there is a nonbeta release as well... are you using something available from beta that is not present in the main?

Reply:
The terraform code itself isn't beta - we just need it to access GCP beta features, like https://cloud.google.com/config-connector/docs/overview. If we don't use those features, there's otherwise no difference.

Reply:
> The terraform code itself isn't beta

Yep, I know :wink:

> we just need it to access GCP beta features

OK, I should have read your comment AND the docs better: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#config_connector_config